ledger/complete/mtrie/trie/trie.go

```diff
@@ -510,18 +528,20 @@ func update(
 	// payload is re-written at a register. CAUTION: we only check that the children are
 	// unchanged. This is only sufficient for interim nodes (for leaf nodes, the children
 	// might be unchanged, i.e. both nil, but the payload could have changed).
-	if !parentNode.IsLeaf() && lChild == lchildParent && rChild == rchildParent {
-		return parentNode, 0, 0, lowestHeightTouched
+	// In case the current node was a leaf, we _cannot reuse_ it, because we potentially
+	// updated registers in the sub-trie
+	if !currentNode.IsLeaf() && newLeftChild == oldLeftChild && newRightChild == oldRightChild {
+		return currentNode, 0, 0, lowestHeightTouched
 	}
 
-	// In case the parent node was a leaf, we _cannot reuse_ it, because we potentially
-	// updated registers in the sub-trie
```

Comment:
The comments at here is confusing as I thought was to explain what is `if prune`. Now moved to the above where we only explain one case for interim node, so that it continues to explain the other case for leaf node.